crates/autopilot/src/decoded_settlement.rs

```diff
@@ -68,13 +78,14 @@ pub struct DecodedTrade {
 }
 
 impl DecodedTrade {
-    fn matches_execution(
-        &self,
-        order: &OrderExecution,
-        tokens: &[Address],
-        domain_separator: &DomainSeparator,
-    ) -> bool {
-        let uid = OrderData {
+    /// Returns the signature of the order.
+    fn signature(&self) -> Result<Signature> {
+        Signature::from_bytes(self.flags.signing_scheme(), &self.signature.0)
+    }
+
+    /// Returns the order uid of the order associated with this trade.
+    pub fn uid(&self, domain_separator: &DomainSeparator, tokens: &[Address]) -> Result<OrderUid> {
```

Comment:
This might be nice to unit test with some example settlements containing interesting order types (EIP1271, PreSign, etc)

Reply:
Good catch. I've added testing the order_uids() in existing unit tests where Eip712 and PreSign are covered.